test/uniport-tests.js

```diff
@@ -0,0 +1,502 @@
+  var send_credentials = test_decl.send_credentials;
+  var wrong_pass = test_decl.wrong_pass;
+  var wrong_user = test_decl.wrong_user;
+
+  var config = {
+    'websocket_ip':  "127.0.0.1",
+    'websocket_port': conf.port,
+    'websocket_ssl': protocol === 'wss',
+    'trace' : configurations.remote_trace
+  };
+
+  var remote = Remote.from_config(config);
+
+  if (require_pass) {
+    remote.on('prepare_subscribe', function(request){
+      if (require_pass && configurations.add_credentials_to_ws_subscribe) {
+        add_credentials_to_request(request.message, conf);
+      };
+      request.once('error', function(e, m){
+        if (!configurations.add_credentials_to_ws_subscribe) {
+          assert.notEqual(e.remote.error,
+                          'forbidden',
+                          'Need credentials for non admin request (subscribe)')
+        };
+      })
+    });
+  };
+
+  remote.once('connected', function() {
+    var b4_ledger_accept = remote._ledger_current_index;
+    var request = new Request(remote, 'ledger_accept');
+
+    if (send_credentials) {
+      add_credentials_to_request(request.message, conf, wrong_pass, wrong_user);
+    }
+
+    request.request(function(e, m){
+      remote.connect(false);
+      var M = function(message) {
+        struct = {port_conf: conf,
+                  request: request.message,
+                  err: e,
+                  response: m,
+                  test_failure: message};
+        return prettyj(struct);
+      }
+
+      if (e != null) {
+        if(should_not_work) {
+          if (require_pass && (!send_credentials || wrong_pass || wrong_user)) {
+            assert.equal(e.remote.error,
+                         'forbidden',
+                         M('should be forbidden'));
+
+          } else if (!require_pass) {
+            assert.equal(e.remote.error,
+                        'forbidden',
+                        M('should be forbidden'));
+          }
+        }
+        assert(should_not_work, // if we got an error it should be expected
+                  /*otherwise*/ M('we should be able to issue admin commands'));
+      }
+      if (m) {
+        if (should_work) {
+          assert.equal((b4_ledger_accept + 1), m.ledger_current_index,
+                       M('admin command should work but did not'));
+        } else {
+          assert.equal(b4_ledger_accept,       m.ledger_current_index,
+                       M('admin command worked but should not have'));
+        }
+      };
+      done();
+    });
+  })
+  remote.connect();
+}
+
+function test_http_admin_command(test_decl, protocol, conf, done)
+{
+  var should_work = test_decl.should_work;
+  var should_not_work = !should_work;
+  var require_pass = conf.admin_password != null;
+  var send_credentials = test_decl.send_credentials;
+  var wrong_pass = test_decl.wrong_pass;
+  var wrong_user = test_decl.wrong_user;
+
+  var url = protocol+'://localhost:'+conf.port + '/';
+
+  var postOptions = {
```

Comment:
So is our naming convention camelCase or under_scores?  Either way, we should be consistent.